### DISCUSSION TITLE
Volume

### DISCUSSION THREAD

**Discussion by cYCL157** [rern/rAudio#37] [Fixed Issues]
I just updated to 202012281 and rebooted.  the volume dial does not really seem to be working right, it was at like 40 before reboot and then after reboot was 100.  on my system, 40 to 100 would make it really loud.  changing up and down seems to have little impact.

**Comment by AndyFx12** [comment]
Hi cYCL157,
please look at MPD settings. There is a new Mixer selector - maybe wrong is selected zu.
No problem with HifiberryDac+ here.

**Reply by cYCL157** [reply]
Thanks @AndyFx12 !  Switching from analog to digital did it -- i am using the same dac as you.  Is there a place to read up on what i should use for mixer device??

**Reply by AndyFx12** [reply]
"Analog" is some kind of an pre-amp. Should be at 100. "Digital" is the "real" mixer.
Andy

**Comment by ceejayemm** [comment]
I am having a similar problem (No Mixer present and 100 Volume) see #26 with a Merus Audio Amp which has worked perfectly upto recently.

Chris

**Comment by rern** [comment]
Show me:
```
aplay -l
aplay -L
amixer scontrols
amixer scontents
```

**Reply by ceejayemm** [reply]
@rern 

As requested:

RAConservatory:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpimerusamp [snd_rpi_merus_amp], device 0: Merus Audio Amp ma120x0p-amp-0 [Merus Audio Amp ma120x0p-amp-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

RAConservatory:~  #  aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
default:CARD=sndrpimerusamp
    snd_rpi_merus_amp, Merus Audio Amp ma120x0p-amp-0
    Default Audio Device
sysdefault:CARD=sndrpimerusamp
    snd_rpi_merus_amp, Merus Audio Amp ma120x0p-amp-0
    Default Audio Device

RAConservatory:~  #  amixer scontrols
Simple mixer control 'A.Mstr Vol',0
Simple mixer control 'B.L Vol',0
Simple mixer control 'C.R Vol',0
Simple mixer control 'D.Lim thresh',0
Simple mixer control 'F.Limiter Enable',0
Simple mixer control 'G.Limiter Attck',0
Simple mixer control 'H.Limiter Rls',0
Simple mixer control 'I.Err flycap',0
Simple mixer control 'J.Err overcurr',0
Simple mixer control 'K.Err pllerr',0
Simple mixer control 'L.Err pvddunder',0
Simple mixer control 'M.Err overtempw',0
Simple mixer control 'N.Err overtempe',0
Simple mixer control 'O.Err pinlowimp',0
Simple mixer control 'P.Err dcprot',0
Simple mixer control 'Q.PM Prof',0
Simple mixer control 'R.Power Mode',0

RAConservatory:~  #  amixer scontents
Simple mixer control 'A.Mstr Vol',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 50
  Mono: 18 [36%] [-32.00dB]
Simple mixer control 'B.L Vol',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 50
  Mono: 50 [100%] [0.00dB]
Simple mixer control 'C.R Vol',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 50
  Mono: 50 [100%] [0.00dB]
Simple mixer control 'D.Lim thresh',0
  Capabilities: volume
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: 0 - 60
  Front Left: 35 [58%] [-15.00dB]
  Front Right: 35 [58%] [-15.00dB]
Simple mixer control 'F.Limiter Enable',0
  Capabilities: enum
  Items: 'Bypassed' 'Enabled'
  Item0: 'Enabled'
Simple mixer control 'G.Limiter Attck',0
  Capabilities: enum
  Items: 'Slow' 'Normal' 'Fast'
  Item0: 'Fast'
Simple mixer control 'H.Limiter Rls',0
  Capabilities: enum
  Items: 'Slow' 'Normal' 'Fast'
  Item0: 'Slow'
Simple mixer control 'I.Err flycap',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Slow'
  Item0: 'Ok'
Simple mixer control 'J.Err overcurr',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Ok'
  Item0: 'Ok'
Simple mixer control 'K.Err pllerr',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Ok'
  Item0: 'Ok'
Simple mixer control 'L.Err pvddunder',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Ok'
  Item0: 'Ok'
Simple mixer control 'M.Err overtempw',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Ok'
  Item0: 'Ok'
Simple mixer control 'N.Err overtempe',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Ok'
  Item0: 'Ok'
Simple mixer control 'O.Err pinlowimp',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Ok'
  Item0: 'Ok'
Simple mixer control 'P.Err dcprot',0
  Capabilities: enum
  Items: 'Ok' 'Error' 'Ok'
  Item0: 'Ok'
Simple mixer control 'Q.PM Prof',0
  Capabilities: enum
  Items: 'PMF0' 'PMF1' 'PMF2' 'PMF3' 'PMF4'
  Item0: 'PMF0'
Simple mixer control 'R.Power Mode',0
  Capabilities: enum
  Items: 'Dynamic power mode' 'Power mode 1' 'Power mode 2' 'Power mode 3'
  Item0: 'Dynamic power mode'

I hope this helps

Chris

**Comment by ceejayemm** [comment]
@rern 

Applied update 20201230 this morning and initially it seems to sort the problems reported above. However after exiting and returning to the MPD page (no reboot) the Mixer Device had returned to 'blank' and the volume was back to 100. Any changes made thereafter were not retained. No change after a reboot either.

Chris

**Comment by ceejayemm** [comment]
@rern 

Still a problem after applying 20201231. I currently have no means of changing the volume on this installation., it is stuck at the last volume level used before this problem arose.

If it helps:

Output Audio:
Device:                Merus Audio Amp
Device Mixer:      Blank
Volume Control:  None - 100%  (0dB) or MPD Software 

This device has work perfectly in earlier RA+Re5/6 installations.

Chris

**Comment by rern** [comment]
Try the UPDATE branch again.

**Comment by ceejayemm** [comment]
@rern 

Applied UPDATE branch but Audio Output now set to:

Output Audio:
Device: Merus Audio Amp
Device Mixer: (Not Available)
Volume Control: None - 100% (0dB) or MPD Software

Still no option to use Hardware mixer.

Chris

**Comment by ceejayemm** [comment]
@rern 

If it helps. On the MPD Audio Output Devices page the Mixer Device box displays '(Not Available)' as noted above. The command box shows the command for this as 'amixer scontrols' which again shows nothing.  However from an ssh session the same command shows:

RAConservatory:~  #   amixer scontrols
Simple mixer control 'A.Mstr Vol',0
Simple mixer control 'B.L Vol',0
Simple mixer control 'C.R Vol',0
Simple mixer control 'D.Lim thresh',0
Simple mixer control 'F.Limiter Enable',0
Simple mixer control 'G.Limiter Attck',0
Simple mixer control 'H.Limiter Rls',0
Simple mixer control 'I.Err flycap',0
Simple mixer control 'J.Err overcurr',0
Simple mixer control 'K.Err pllerr',0
Simple mixer control 'L.Err pvddunder',0
Simple mixer control 'M.Err overtempw',0
Simple mixer control 'N.Err overtempe',0
Simple mixer control 'O.Err pinlowimp',0
Simple mixer control 'P.Err dcprot',0
Simple mixer control 'Q.PM Prof',0
Simple mixer control 'R.Power Mode',0

So it would seem that the web app is not seeing the same information as the basic ssh command.

Chris

**Comment by rern** [comment]
Try:
`sed -i '52 s/pvolume/volume/' /srv/http/bash/mpd-devices.sh`

**Reply by ceejayemm** [reply]
That seems to have done the trick. The Mixer Device is now 'A.Mstr.Vol' and I can select 'Mixer Device' as the Volume Control.

Out of interest though, the web app Mixer Device / amixer scontrols dialogue box is still empty.

Thanks

Chris
